ge/org.osate.ge.swt/src/org/osate/ge/swt/prototypes/PrototypeEditor.java

```diff
@@ -0,0 +1,321 @@
+ * files ("Third Party Software"). The Third Party Software that is used by this program is dependent upon your system
+ * configuration. By using this program, You agree to comply with any and all relevant Third Party Software terms and
+ * conditions contained in any such Third Party Software or separate license file distributed with such Third Party
+ * Software. The parties who own the Third Party Software ("Third Party Licensors") are intended third party benefici-
+ * aries to this license with respect to the terms applicable to their Third Party Software. Third Party Software li-
+ * censes only apply to the Third Party Software and not any other portion of this program or this program as a whole.
+ */
+package org.osate.ge.swt.prototypes;
+
+import java.util.Objects;
+import java.util.function.Consumer;
+import java.util.stream.Stream;
+
+import org.eclipse.jface.layout.GridDataFactory;
+import org.eclipse.jface.layout.GridLayoutFactory;
+import org.eclipse.swt.SWT;
+import org.eclipse.swt.custom.CLabel;
+import org.eclipse.swt.widgets.Composite;
+import org.eclipse.swt.widgets.Control;
+import org.osate.ge.swt.ChangeEvent;
+import org.osate.ge.swt.EventSource;
+import org.osate.ge.swt.check.CheckboxEditor;
+import org.osate.ge.swt.check.CheckboxEditorModel;
+import org.osate.ge.swt.internal.InternalUtil;
+import org.osate.ge.swt.name.NameEditor;
+import org.osate.ge.swt.name.NameEditorModel;
+import org.osate.ge.swt.selectors.ComboSelector;
+import org.osate.ge.swt.selectors.FilteringListSelectorField;
+import org.osate.ge.swt.selectors.LabelFilteringListSelectorModel;
+import org.osate.ge.swt.selectors.RadioSelector;
+import org.osate.ge.swt.selectors.SelectorModel;
+
+/**
+ * View for editing a prototype.
+ *
+ */
+public final class PrototypeEditor<C> extends Composite {
+	private static final String WIDGET_ID_PREFIX = "org.osate.ge.swt.prototypes.prototypeEditor.";
+	public static final String WIDGET_ID_CHOOSE_CLASSIFIER_BUTTON = WIDGET_ID_PREFIX + "chooseClassifier";
+	public static final String WIDGET_ID_CLASSIFIER_LABEL = WIDGET_ID_PREFIX + "classifierButton";
+	public static final String WIDGET_ID_TYPE_COMBO = WIDGET_ID_PREFIX + "type";
+
+	private final PrototypeEditorModel<C> model;
+	private final CheckboxEditor refinedEditor;
+	private final Control directionLabel;
+	private final RadioSelector<PrototypeDirection> directionEditor;
+	private final CheckboxEditor arrayFlagEditor;
+	private final Consumer<ChangeEvent> changeListener = e -> refresh();
+
+	public PrototypeEditor(final Composite parent, final PrototypeEditorModel<C> model) {
+		super(parent, SWT.NONE);
+		this.model = Objects.requireNonNull(model, "model must not be null");
+		InternalUtil.setColorsToMatchParent(this);
+		this.setLayout(GridLayoutFactory.fillDefaults().numColumns(2).create());
+
+		//
+		// Refinement
+		//
+		refinedEditor = new CheckboxEditor(this, new CheckboxEditorModel() {
```

Comment:
I wasn't sure whether to put this comment on CheckboxEditor or here, but the issue is that the Refine checkbox and Array checkbox is filling the horizontal space in the property view and it is possible to click the checkbox even when the mouse is not on the button.  Might just need a composite container? Not sure of the best approach.

I have attached a screenshot with an arrow pointing to where the mouse can be and still click the Refine or Array checkboxes.

![refine](https://user-images.githubusercontent.com/8505197/80969347-0e939d80-8ddf-11ea-8ca4-ade480f15601.png)

Reply:
Fixed in new push.

Reply:
Refine is fixed, Array checkbox seems to still have the old behavior.